SMLHelper/Options/ChoiceAttribute.cs

```diff
@@ -0,0 +1,69 @@
+﻿using System;
+using System.Linq;
+
+namespace SMLHelper.V2.Options
+{
+    /// <summary>
+    /// Attribute used to signify the specified member should be represented in the mod's options menu as a
+    /// <see cref="ModChoiceOption"/>. Works for either <see cref="int"/> index-based, <see cref="string"/>-based, or
+    /// <see cref="Enum"/>-based members.
+    /// </summary>
+    /// <remarks>
+    /// <see cref="Enum"/> choices can also be parsed from their values by merely omitting the <see cref="ChoiceAttribute"/>.
+    /// </remarks>
+    /// <example>
+    /// <code>
+    /// using SMLHelper.V2.Json;
+    /// using SMLHelper.V2.Options;
+    /// 
+    /// public enum CustomChoice { One, Two, Three }
+    /// 
+    /// [Menu("My Options Menu")]
+    /// public class Config : ConfigFile
+    /// {
+    ///     [Label("My index-based choice"), Choice("One", "Two", "Three")]
+    ///     public int MyIndexBasedChoice;
+    ///     
+    ///     [Label("My enum-based choice")]
+    ///     public CustomChoice MyEnumBasedChoice;
+    /// }
+    /// </code>
+    /// </example>
+    /// <seealso cref="MenuAttribute"/>
+    /// <seealso cref="LabelAttribute"/>
+    /// <seealso cref="ModChoiceOption"/>
+    /// <seealso cref="Json.ConfigFile"/>
+    [AttributeUsage(AttributeTargets.Property | AttributeTargets.Field, AllowMultiple = false)]
+    public sealed class ChoiceAttribute : Attribute
+    {
+        /// <summary>
+        /// The list of options that will be displayed.
+        /// </summary>
+        public string[] Options { get; }
```

Comment:
Remind me: Does this have to be an array? Or could it be an `IEnumerable<string>`?

Reply:
The pre-built `ModChoiceOption` takes either an `object[]` array or `string[]` array in the constcutor so it will be coerced to that in the end regardless, so I declared it as one here.

Reply:
https://github.com/SubnauticaModding/SMLHelper/blob/master/SMLHelper/Options/ChoiceModOption.cs#L88

Reply:
@AlexejheroYTB  Do you think it makes sense to update all these array parameters to `IEnumerable`s?  
Was there a reason they were arrays?

Reply:
Trawling through the code-base, it looks like eventually it leads here:
https://github.com/SubnauticaModding/SMLHelper/blob/master/SMLHelper/Options/ChoiceModOption.cs#L157
```cs
    public class ModChoiceOption : ModOption
    {
        /// <summary>
        /// The array of readable string options to choose between in the <see cref="ModChoiceOption"/>.
        /// </summary>
        public string[] Options { get; }
```

And from finding all references in VS, that property is used when called out to this method within `Assembly-CSharp.dll`:
`public uGUI_Choice AddChoiceOption(int tabIndex, string label, string[] items, int currentIndex, UnityAction<int> callback = null);`, passed as the `string[] items` parameter, as well as used for the `UnityAction`:

```cs
var choice = panel.AddChoiceOption(tabIndex, Label, Options, Index,
    new UnityAction<int>((int index) => parentOptions.OnChoiceChange(Id, index, Options[index])));
```

So it'll end up being coerced down to the a `string[]` array in the end.

Reply:
Then I guess we can leave the explicit string array. No sense in upending everything over such a small detail.